packages/react/src/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,357 @@
+import React, {
+  Children,
+  isValidElement,
+  cloneElement,
+  useRef,
+  useCallback,
+  useState,
+  useEffect,
+  PropsWithChildren,
+  forwardRef,
+  memo,
+  Ref,
+  ReactElement
+} from 'react'
+import {Button, Text} from '../'
+import {useAnchoredPosition} from '../hooks/useAnchoredPosition'
+import {useOnClickOutside} from '../hooks/useOnClickOutside'
+import {useKeyboardEscape} from '../hooks/useKeyboardEscape'
+
+import {default as clsx} from 'clsx'
+import {CheckIcon, ChevronDownIcon} from '@primer/octicons-react'
+import {FocusKeys, focusZone} from '@primer/behaviors'
+import type {BaseProps} from '../component-helpers'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/action-menu/colors-with-modes.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './ActionMenu.module.css'
+
+const testIds = {
+  root: 'ActionMenu',
+  get button() {
+    return `${this.root}-button`
+  },
+  get overlay() {
+    return `${this.root}-overlay`
+  },
+  get list() {
+    return `${this.root}-list`
+  },
+  get item() {
+    return `${this.root}-list-item`
+  }
+}
+
+type ActionMenuProps = {
+  /**
+   * The content of the ActionMenu. Must be an ActionMenu.Button and an ActionMenu.Overlay
+   */
+  children:
+    | ReactElement<ActionMenuButtonProps>
+    | ReactElement<ActionMenuOverlayProps>
+    | Array<ReactElement<ActionMenuButtonProps> | ReactElement<ActionMenuOverlayProps>>
+  /**
+   * Determines whether the ActionMenu is disabled
+   */
+  disabled?: boolean
+  /**
+   * Determines whether the ActionMenu is open by default
+   */
+  open?: boolean
+  /**
+   * Callback that is called when the open state changes
+   */
+  onOpenChange?: (open: boolean) => void
+  /**
+   * Callback that is called when an item is selected.
+   * The value of the selected item is passed as an argument.
+   */
+  onSelect?: (newValue: string) => void
+  /**
+   * The selection variant of the ActionMenu.
+   * - `single`: Only one item can be selected at a time. Menu items as a menuitemradio.
+   * - `none`: No items can be selected. Menu items as a menuitem.
+   */
+  selectionVariant?: 'single' | 'none'
```

Comment:
should this be `none` by default?

Reply:
Good question and thanks for bringing this up. I'm inclined to agree, because this is now an ActionList. The original intent for this component however, was to behave as a SelectMenu first, which is why it's still defaulting to single.